### ISSUE TITLE
How to update craco-antd for the latest craco?

### ISSUE THREAD

**Issue by ndbroadbent** [gsoft-inc/craco#115]
Hello,

I'm working on an update for `craco-less` and `craco-antd`, but I'm getting stuck at this point:

```
// In lib/craco-antd.js
module.exports = {
  overrideCracoConfig: ({ cracoConfig }) => {
    if (!cracoConfig.babel) cracoConfig.babel = {};
    if (!cracoConfig.babel.plugins) cracoConfig.babel.plugins = [];
    cracoConfig.babel.plugins.push([
      "import",
      { libraryName: "antd", libraryDirectory: "es", style: true }
    ]);
    return cracoConfig;
  }
};
```

This no longer seems to be working in the latest `craco`. When I inspect the generated webpack config, I can't see any reference to `libraryName` or `libraryDirectory` in the JSON. Has something changed with `cracoConfig` or the `babel.plugins` array? Thanks!

**Comment by patricklafrance** [comment]
Hi @ndbroadbent 

I don't remember that craco had specific support for `libraryName` or `libraryDirectory`. Maybe something changed in CRA?

In version 5.4.0 a fix has been released for plugins, maybe this is what affects your plugin?

For more info view issue https://github.com/gsoft-inc/craco/issues/101

Thank you,

Patrick

**Comment by ndbroadbent** [comment]
Hello, sorry yes the `libraryName` keywords were just an example from my usage, because I was trying to find them in the generated webpack config, but they were no longer present. E.g. I was trying to see if the configuration had moved somewhere else.

I don't think #101 would cause this issue. Does my code look right in the example? How would you add the babel `import` plugin with some options? Thanks for your help!

**Comment by patricklafrance** [comment]
How do you assert they are no longer present @ndbroadbent ?

I made and a test and outputed the webpack config using the craco api feature, which use basically the same pipeline that the standard craco config override feature but return an object instead of overriding the CRA config and I got this:

```javascript
// output

"plugins": [
                "babel-plugin-jsx-control-statements",
                "babel-plugin-react-require",
                [
                  "babel-plugin-react-css-modules",
                  {
                    "generateScopedName": "[local]___[hash:base64:5]",
                    "attributeNames": {
                      "activeStyleName": "activeClassName"
                    }
                  }
                ],
                [
                  "@babel/plugin-proposal-decorators",
                  {
                    "legacy": true
                  }
                ],
                [
                  "import",
                  {
                    "libraryName": "antd",
                    "libraryDirectory": "es",
                    "style": true
                  }
                ]
              ]
```

To achieve this config, I added your plugin to my craco config:

```javascript
// craco.config.js

    plugins: [
        {
            plugin: {
                overrideCracoConfig: ({ cracoConfig }) => {
                    if (!cracoConfig.babel) cracoConfig.babel = {};

                    if (!cracoConfig.babel.plugins) cracoConfig.babel.plugins = [];

                    cracoConfig.babel.plugins.push([
                      "import",
                      { libraryName: "antd", libraryDirectory: "es", style: true }
                    ]);
                    
                    return cracoConfig;
                },
```

Thank you,

Patrick

**Comment by ndbroadbent** [comment]
Hello,

Yes I can see that the craco config is being updated correctly, but the resulting webpack configuration doesn't include this change.

* [This is where I'm returning `overrideCracoConfig` for the plugin](https://github.com/DocSpring/craco-antd/blob/master/lib/craco-antd.js#L78-L94)

Then I have these [tests for both development and production](https://github.com/DocSpring/craco-antd/blob/master/lib/craco-antd.dev.test.js), to ensure that the webpack configuration is modified correctly.

After updating craco to the latest version, the "import" plugin seems to have stopped working. [This code](https://github.com/DocSpring/craco-antd/blob/master/lib/craco-antd.dev.test.js#L96-L103) checks the JS rule in the webpack config, and ensures that it contains the correct elements:

```
expect(jsRule.options.plugins[0]).toEqual([
    "import",
    { libraryName: "antd", libraryDirectory: "es", style: true }
  ]);
```

But this has started failing now. The configuration has changed to this:

```
expect(jsRule.options.plugins[0][0]).toContain(
    `${path.sep}babel-plugin-named-asset-import${path.sep}index.js`
  );
  expect(jsRule.options.plugins[0][1]).toEqual({
    loaderMap: {
      svg: {
        ReactComponent: "@svgr/webpack?-svgo,+titleProp,+ref![path]"
      }
    }
  });
```

I couldn't figure out what had happened to my babel import plugin, so I was running `JSON.stringify(webpackConfig)`, and then searching the JSON for keywords like `libraryName`, `libraryDirectory`, and `antd`, but I couldn't find these anywhere in the webpack config.

I have a WIP commit to update the `craco-antd` tests, but the `craco-less` tests are currently passing. [Here is the "test harness" I'm using to set up the craco config and generate the webpack config.](https://github.com/DocSpring/craco-less/blob/master/lib/craco-less.dev.test.js#L14-L38)

Basically I call `loadWebpackDevConfig`, then `applyCracoConfigPlugins` and `applyWebpackConfigPlugins`. Then I run some tests for the generated webpack config, to make sure everything is configured properly.

It looks like the `overrideWebpackConfig` functions calls are still working properly, but I'm struggling to figure out what has changed in my `overrideCracoConfig` function, since the `cracoConfig.babel.plugins` array seem to be disappearing somewhere.

Thanks a lot for your help!

**Comment by ndbroadbent** [comment]
Good news! @Vovan-VE has figured out how to fix the failing tests for craco-antd: https://github.com/DocSpring/craco-antd/pull/23

There was a problem in my test cases. I was previously calling `applyCracoConfigPlugins` and `applyWebpackConfigPlugins`, but this did not produce the correct webpack config. It looks like the correct way is to call the `mergeWebpackConfig` function, and this produces the webpack config that includes the Babel import rule for antd. Thanks for your help!

**Comment by patricklafrance** [comment]
Awesome, thank you @Vovan-VE !

That's great news!
